### ISSUE TITLE
Error on first startup

### ISSUE THREAD

**Issue by samyoulmane** [debanjum/khoj#130]
Hello !

I've just installed khoj with pip and when I want to run the app with `khoj --no-gui`, I get this error :

```bash
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/bin/khoj", line 5, in <module>
    from src.main import run
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/src/main.py", line 23, in <module>
    from src.configure import configure_server
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/src/configure.py", line 10, in <module>
    from src.processor.ledger.beancount_to_jsonl import BeancountToJsonl
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/src/processor/ledger/beancount_to_jsonl.py", line 8, in <module>
    from src.processor.text_to_jsonl import TextToJsonl
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/src/processor/text_to_jsonl.py", line 16, in <module>
    class TextToJsonl(ABC):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/src/processor/text_to_jsonl.py", line 21, in TextToJsonl
    def process(self, previous_entries: list[Entry]=None) -> list[tuple[int, Entry]]: ...
TypeError: 'type' object is not subscriptable
```

Any idea how to fix it ?

I'm on macOS Ventura 13.1 and Python 3.8.2

Thank you !

**Comment by debanjum** [comment]
Hey @samyoulmane, thanks for reporting this issue (with version details). I can repro it on my end with python 3.8 as well. It seems like a typing issue. 
1. To mitigate this issue, you can upgrade to python 3.9
2. Or you can wait a few days for me to push out a fix for this

**Comment by samyoulmane** [comment]
Thank you. I upgraded to python 3.9 and it worked perfectly !

**Comment by BasilPH** [comment]
I installed `khoj-assistant` on Python 3.8.2 and still get `TypeError: 'type' object is not subscriptable` error. I've also tried `pip install --upgrade --pre khoj-assistant` as per #140, but that didn't help.

I ended up install khoj in a virtualenv with python 3.9, which worked. So this isn't blocking me. I just wanted to let you know and I appreciate that you're building Khoj 🙏

**Comment by debanjum** [comment]
Thanks @BasilPH, for letting us know. Glad you got it working with a virtualenv in python 3.9!

We should update the [documentation](https://docs.khoj.dev/#/setup) to indicate that Khoj only supports python version >=3.9. 

For context: Some of Khoj's dependencies (gpt4all?) don't support python 3.8 anymore so we dropped support for it and moved our type hinting to use the newer styles available from python 3.9+
